src/daft-plan/src/physical_plan.rs

```diff
@@ -410,6 +421,8 @@ impl PhysicalPlan {
             Self::InMemoryScan(..) => vec![],
             Self::TabularScan(..) | Self::EmptyScan(..) => vec![],
             Self::Project(Project { input, .. }) => vec![input.clone()],
+            #[cfg(feature = "python")]
+            Self::ActorPoolProject(ActorPoolProject { input, .. }) => vec![input.clone()],
```

Comment:
I think we might also have to add `ActorPoolProject` as a case to `with_new_children` below.

Reply:
And lets remove the catch all at the bottom!